oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/type/StorageBuilder.java

```diff
@@ -16,18 +16,16 @@
  *
  */
 
-package org.apache.skywalking.oap.server.core.storage;
-
-import java.util.Map;
+package org.apache.skywalking.oap.server.core.storage.type;
 
 /**
- * Converter between the give T and Map.
+ * Converter between the give T and K.
  *
  * @param <T> A storage entity implementation.
+ * @param <K> The data structure recognized by the storage implementation.
  */
-public interface StorageBuilder<T extends StorageData> {
-
-    T map2Data(Map<String, Object> dbMap);
+public interface StorageBuilder<T, K> {
+    T storage2Entity(K dbMap);
```

Comment:
Also get rid of the map at interface definition?

```suggestion
    T storage2Entity(K storageData);
```